File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/CharmRecord.java
package kz.greetgo.sandbox.controller.model;

public class CharmRecord {
  public Integer id;
  public String name;
  public String description;
  public float energy;

  public CharmRecord() {}

  public CharmRecord(Integer id, String name, String description, float energy) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.energy = energy;
  }
}

File: sandbox.client/front/ts/model/CharmRecord.ts
export class CharmRecord {
  public id: number;
  public name: string;
  public description: string;
  public energy: number;

  public assign(o: any): CharmRecord {
    this.id = o.id;
    this.name = o.name;
    this.description = o.description;
    this.energy = o.energy;
    return this;
  }

  public static copy(a: any): CharmRecord {
    let ret = new CharmRecord();
    ret.assign(a);
    return ret;
  }
}